### PR TITLE
Removes Targeting Mode

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Utterly guts and removes every single trace (that I could find) of targeting mode.

No ugly on-screen icons, no code, no more buggy overlays getting stuck on simple animals, no more "I win button", no more ....ANYTHING related to targeting mode.

Yay for more HUD space and more skill when using guns. Sorry, if you relied on this to win in combat, you're going to have to up your game (and aim).

@Spacemanspark

### CODE REVIEW

**Comment by Aurorablade** [comment]
Eh, I never really used it as a 'i win' and more of a 'stop your shit or i am gonna shoot you' function as a warning measure...but its all good!

**Comment by Robertha89** [comment]
I never used it to begin with cause i was not a lil shit, and if i needed to aim to shoot, then aimmode fired too slow, why use aimmode when you can empty your laser between those shots.

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -268,7 +268,6 @@
 			return 0
 
 		move_delay = world.time//set move delay
-		mob.last_move_intent = world.time + 10
```
Comment (tigercat2000) [inline code/modules/mob/mob_movement.dm]: um, how is this related to targeting mode

Comment (Fox-McCloud) [inline code/modules/mob/mob_movement.dm]: It's utilized quite extensively in projectiles/targeting.dm

**Comment by Spacemanspark** [comment]
YAY.
:+1:

**Comment by Allfd** [comment]
:+1: 

I liked the idea of being able to easily alert someone you were targeting them.  But it is basically just a big easy button.

**Comment by Necaladun** [comment]
Completely against.

Combat based on rapid clicking is boring, and not the point of SS13. This is meant to be an RPG, not a 2d click-combat game. Combat being decided by who can click the best and have the lowest ping is lame.

Targetting well simulated having your gun drawn and aimed at someone. Especially useful for security and bartenders. It's a way of simulating IC behaviour without relying on OOC factors such as lag or ability to click.

Calling it an "i win" button is just false, there are plenty of times it does not work that way. If it's too powerful, there are plenty of ways of balancing it. Slow the fire rate, give it a miss chance, etc.

If it was such an I win button, it would be used a hell of a lot more. Combat would be dominated by people using it constantly. That is not the case, at all. 

Additionally, it is one of the only way to have a hostage or escalate a combat where people can still talk. This allows plenty of times for RP situations, whether it's the bartender telling someone to fuck off, or sec telling you not to move.

Removing this just means your only option is to shoot first.

**Comment by tigercat2000** [comment]
@Necaladun It never misses. It requires one click to lock onto someone, and if they move at all, they are hit. No chance to run or get out of the way. 

Hostage situations don't need this. Hostage situations need people paying attention and more importantly, talking. You don't need to aim at someone to tell them to fuck off with a laser out.

**Comment by CrAzYPiLoT-SS13** [comment]
I completely agree with Neca on this one. Tiger, have you read his comment?
It does not currently give a chance of missing, but as he said it can be
balanced out by adding the said chance and many other ways.
@ZomgPonies

**Comment by CrAzYPiLoT-SS13** [comment]
Also, maybe add some kind of a delay to locking on while still moving behind the target?

**Comment by tigercat2000** [comment]
@CrAzYPiLoT-SS13 I did read his comment. I disagree with his view. 

This is compounded by the fact that the system is a glitchy mess at the best of times, and I don't like glitchy messes.

Could the system be balanced? Maybe. Is it worth keeping at all, 'balanced' or not? No, not in my opinion.

**Comment by Necaladun** [comment]
@tigercat2000 - if they move. Simple counter is not moving. Or moving behind a wall, locker, or one of their friends.

It's much more of a defensive thing - if you wanna kill someone, you click spam them. 

Again, if this is so problematic - why is it not used exclusively?

**Comment by Robertha89** [comment]
I have avoided it on several occasions, the best way, move someone else in the way, then run, and dodge outta the way. or if you are feeling clever, build a computer frame, and drag it behind you.
Even so, the main use for it is really RP! + the fact that it does not mean good ping = win.

**Comment by tigercat2000** [comment]
@Robertha89 It all depends on situation. If you are in the middle of an open space with no cover and someone targets you, you are 100% fucked. If they decide to target you in the bar with plenty of places to move behind, or while you are with your buddies, then yeah, I suppose you can defend against it.
The problem comes from the fact that it cannot miss. It is quite literally hooked into movement, you don't need any amount of skill or a fair response time to use it. 

This is considered a bad thing. It retracts from requiring an amount of skill while playing security.

@Necaladun You make the point that this is meant to be an RPG. I'm relatively sure there are a number of players who would disagree that it plays anything like one.

**Comment by tigercat2000** [comment]
And, now, if you don't mind, I regret getting involved in this.
@Fox-McCloud feel free to defend your PR, I'm out.

**Comment by Necaladun** [comment]
I get the dislike of it being 100% hit chance if they move in the open being disliked - but that seems rather to simulate how it would happen if you have a gun drawn on you. Balancing it by fire delays or miss chances seem the best idea.

I just havn't seen the problems it creates for such an "I win button".

**Comment by Regen1** [comment]
I completely agree with Neca on this one, fuck off with your stupid "this is a game like call of duty". It's an RPG game.
If someone aims a gun at you then fucking stop whatever you are doing, are you a traitor? Well that's fucking shitty cause you just got caught red handed, are you a wizard? Well fuck shit you got spells to literally turn invisible and invincible to get the fuck away. Are you a Nuke ops? Fuck son, you have a sub machine gun at worst, suck it up and return fire.

Point is, I am getting tired of balancing things that makes this more into a "balanced combat game" instead of an RP game, in an RP game people have advantages and disadvantages, deal with it.

Game is not user friendly.

**Comment by CrAzYPiLoT-SS13** [comment]
Regen, the completely pointstabbing person. Only problem I see in your
post, Regen, is that New Cops can't get a gun out or use it with the "don't
allow item use" thingie.

**Comment by Robertha89** [comment]
@tigercat2000 Well, I have personally avoided such in the hallways, thanks to dodging, other people, and running outta range, and even then, that is an issue for mostly non-antags, as most antags have ranged weapons they can use to return fire faster then you can shoot in aimed mode, pluss if you stand still there is no issue at all.
Was this not nerfed a while back where it stopped forcing people to walk?

**Comment by Spacemanspark** [comment]
Not nerfed enough.

**Comment by Necaladun** [comment]
The force-to-walk was dumb for sure, good that was removed.

But without aim mode, if someone has a knife of the like out, there is little you can do to talk to them to make them drop it without risking them running at you/away. Typing to talk stops you shooting, so without aim mode, you'd best just shoot first. 

Without tools like this, we become more and more of a click-based-combat game. And in the end, SS13 makes a pretty shitty game. 

If this were highly abused or used heavily, I could see the problems. But entire rounds go by without aim mode being used at all.

**Comment by Fox-McCloud** [comment]
When the game automates process for you that you normally have to do, yourself, it's an "I win button"; it doesn't matter if the game is RP oriented, combat oriented, or a mixture of the two. You're having the game pre-calculate the distance and relative angle of a person and to fire the SPLIT second that the game engine registers they move, all for you. 

This is handholding, plain and simple and is designed for a ultra-high RP environment in mind (which we don't have) where you're *expected* to never move if you're targeted, to never fire outside of this mode, and otherwise have to approach the situation "realistically". 

Compensating for the lack of ability, in any game, is poor game design and is something that should be deliberately avoided at all costs. The player should always be the one that has agency over his own actions; not a game engine that pre-calcuates things for you.

It's a terrible feature that junks up the UI (its present for everyone, whether theyr'e using a gun or not).

**Comment by Necaladun** [comment]
Again - if it's such an "I win" button then I'd expect to see it used constantly. Round after round goes by without it being used. 

The fact that you can't shoot while "talking" is a great enough reason to have it. If you're targetted by anything except a taser or other instant stun weapon, running away and taking the damage is an option. Those weapons allow you to click once to win a fight pretty easily - using aim mode on them actually gives more options to the person being targetted. You still have to click on the person to target them in the first place.

There are numerous counters to it as well, unless you're completely in the open. And if you're completely in the open with someone pointing a gun at you, then you're fucked. I see nothing wrong with that.

I'll ask that this is marked controversial so that more thought can be put into it. A better solution atm would be about adding a larger delay to it, etc.

**Comment by Fox-McCloud** [comment]
- If that's really the case, as you claim, then removing it definitely doesn't do any harm, what so ever.

- Yes, you have to click once, then all the aiming is done for you, vs having to adjust your aim after each shot; this is easy-moding things and having an "I win" button mo matter how you slice it--it would be akin to having you auto-punch and auto-target the person being bunched if you can get just a single click on them and stay in range.

- There's no counter to it; once you're clicked on, you're targeted and at the complete whims of the person holding the gun---even if they have a 2 second reaction time, they still fire the split second you move (or talk into the gosh damn radio).

- It's not that controversial in the communities eyes, as this has been brought up a few times, and...again, the previous PR garnered *overwhelming* support. for it.

**Comment by Necaladun** [comment]
If numerous people in this thread and a headmin are dissenting the idea, then it is controversial and needs more discussion.

**Comment by Fox-McCloud** [comment]
Supporters:
- @TheDZD 
- @Fox-McCloud
- @ZomgPonies 
- @Allfd 
- @Spacemanspark 
- @FreeStylaLT 
- @Aurorablade 
- @veganzombeh 
- @Robertha89  
- @tigercat2000 
- @fj45 

Opponents
- @Necaladun 
- @Regen1 
- @CrAzYPiLoT-SS13 


Not that controversial, Neca.

**Comment by tigercat2000** [comment]
@Fox-McCloud I have to disagree there.
There are supporters.
There are opponents.
That means there is a controversy issue here. It's worth letting more people put in their opinions.

**Comment by Necaladun** [comment]
I'm not saying it's super controversial, I'm saying valid points have been raised against it. These need to be fleshed out. 

It is not an urgent issue that needs to be merged ASAP.

**Comment by Necaladun** [comment]
"The problem comes from the fact that it cannot miss. It is quite literally hooked into movement, you don't need any amount of skill or a fair response time to use it."

I'm really not sold on the idea that clicking on a sprite is a good use of "skill". 

"There's no counter to it"

Numerous counters have been shown - getting behind cover, getting behind someone else, stunning them first, etc. 

"once you're clicked on, you're targeted and at the complete whims of the person holding the gun-"

This is the same as if someone clicks on you once with a taser without aim mode.

"even if they have a 2 second reaction time, they still fire the split second you move"

This is a good equalizer for lag.

**Comment by Fox-McCloud** [comment]
- Like it or not, there's not a whole lot of expression of "skill" in SS13 that isn't based on exploitation of burden of knowledge; this is probably the way it will always, but we shouldn't work at making the game easy-mode or removing the few skill caps that are in place.

- You click once, and are targeted; there's no way to avoid that. Once targeted, the person is going to *automatically fire* if you so much as move --likewise even merely talking into your radio is going to trigger and update for being targeted.

Ducking behind cover doesn't prevent you from being targeted; it just prevents the projectile from hitting--just like if you had to actually aim.

- That's not an apt or good comparison *at all*. The person who got hit with the taser had to be HIT first; the person had to actually aim to hit you--and that's his reward for doing so. Targeting removes that and just hands a straight up reward for being able to click on the person...once.

- So are hitscan projectiles and aimbots; those are both hardly good things. Taking skill *away* from something is never a good thing, especially in a game that isn't 100% roleplay (if it was, then that could hypothetically be a different story).

**Comment by Necaladun** [comment]
"-likewise even merely talking into your radio is going to trigger and update for being targeted."

- that's stupid, and should be removed for sure. It seems against the intent and an oversight.

"The person who got hit with the taser had to be HIT first; the person had to actually aim to hit you"

I'm not sure how this is different from targetting? To taze someone, you have to click on them. Or behind them, or infront of them. That's much easier than clicking directly on them.

Adding a firing delay to aim mode seems a better solution here.

"Taking skill away from something is never a good thing"

If botany required you to do complex calculations, while writing a limerick, taking skill away would be great. It'd be the wrong skill.

The skills involved in SS13 should be more about psychology, conversation, roleplay, etc. "Clicking on sprites" is a pretty crappy skill IMO. It's a

"especially in a game that isn't 100% roleplay (if it was, then that could hypothetically be a different story)."

This seems to be something that will lower the potential for and the amount of roleplay further. This is why I'm against it. Keeping it will not make us suddenly "super high-RP". Removing it won't make us no-RP, but it will hurt the people who want to be able to RP during a tense moment. The way that talking stops you from being able to act severely restricts this.

**Comment by Spacemanspark** [comment]
Put an overlay, perhaps.
Doesn't autofire.
But it does give a nice message of 'HEY YOU THERE PAY ATTENTION I'VE GOT A GUN TRAINED ON YOU.'

**Comment by Necaladun** [comment]
The psychological element is great there, but it'd end up being used even less without mechanical effect. 

If you've already clicked, you could have shot them. As it's meant to represent you having trained your gun on them, perhaps having it fire only once if they move/shoot, then auto return you to shoot mode would work?

This would reward the person who clicks first. Two people aiming at each other in a Mexican standoff would be awesome too.

Alternatively, have a fire delay that increased with each successive shot. Harder to code though.

**Comment by CrAzYPiLoT-SS13** [comment]
I agree with both Spaceman and Neca, but I'm kinda leaning towards Spaces
idea. This could be done so that if you want to shoot them if they moved,
just point the friggin mouse pointer  either behind or in front of them. A
new thing to learn. A new skill if it can be said so.

**Comment by Fox-McCloud** [comment]
Since the forums is down, I do have an idea that would leave the ability to "get people's attention" intact while still killing off the shitty aimbot portion of this.

**Comment by CrAzYPiLoT-SS13** [comment]
And that is?

**Comment by ZomgPonies** [comment]
:-1:  Put me in the dissenters. I'd rather see it balanced than removed, as it has definite RP usage and it's raw combat potential is pretty limited.

**Comment by tinfoiltophat** [comment]
In my OPINION, if you really don't like aim mode, just remove the "YOU MOVE YOU DIE" part, and keep the ability to actually aim at people for RP and  hostage taking shit.

**Comment by Fox-McCloud** [comment]
Don't feel like working on a feature that I inherently view as cheating, sorry.

**Pull request closed without merging**